### PR TITLE
footwraps now go tap tap tap

### PR DESCRIPTION
**Pull request by cuackles**
# Document the changes in your pull request

Footwraps now do the barefoot sound cus most of the feet that is touching the floor its not covered by the wrap done almost entirely by mqiib
# Changelog

footwraps now do the barefoot sound 

:cl:    
tweak: tweaked a few things  
/:cl:

### CODE REVIEW

**Comment by JamieD1** [comment]
I mean this doesnt make sense "Most of the foot is not touching the ground" then why does it stop tables and glass? You are opening a can of worms hear, however I will settle if the sound is half as loud as the bare foot one.

**Comment by Mqiib** [comment]
> I mean this doesnt make sense "Most of the foot is not touching the ground" then why does it stop tables and glass? You are opening a can of worms hear, however I will settle if the sound is half as loud as the bare foot one.

Lizards actually have a different sound than humans when barefoot, being the claws hitting the floor rather than the whole foot slapping down. Footwraps don't totally cover the claws so it'd make sense that they would still be heard while walking. There's no reason for them to be muted at all, and the way it works currently wearing footwraps makes it sound like you're walking around in heavy boots which makes even less sense than what this pr does. As for the glass argument, the soft parts of the foot are still covered while the claws, which are hard and not particularly sensitive, are the only things that would come in contact with the glass.

**Comment by JamieD1** [comment]
> > I mean this doesnt make sense "Most of the foot is not touching the ground" then why does it stop tables and glass? You are opening a can of worms hear, however I will settle if the sound is half as loud as the bare foot one.
> 
> Lizards actually have a different sound than humans when barefoot, being the claws hitting the floor rather than the whole foot slapping down. Footwraps don't totally cover the claws so it'd make sense that they would still be heard while walking. There's no reason for them to be muted at all, and the way it works currently wearing footwraps makes it sound like you're walking around in heavy boots which makes even less sense than what this pr does. As for the glass argument, the soft parts of the foot are still covered while the claws, which are hard and not particularly sensitive, are the only things that would come in contact with the glass.

What about the other species, its not all about lizards you know?

**Comment by cuackles** [comment]
> > > I mean this doesnt make sense "Most of the foot is not touching the ground" then why does it stop tables and glass? You are opening a can of worms hear, however I will settle if the sound is half as loud as the bare foot one.
> > 
> > Lizards actually have a different sound than humans when barefoot, being the claws hitting the floor rather than the whole foot slapping down. Footwraps don't totally cover the claws so it'd make sense that they would still be heard while walking. There's no reason for them to be muted at all, and the way it works currently wearing footwraps makes it sound like you're walking around in heavy boots which makes even less sense than what this pr does. As for the glass argument, the soft parts of the foot are still covered while the claws, which are hard and not particularly sensitive, are the only things that would come in contact with the glass.
> 
> What about the other species, its not all about lizards you know?

It also affects polys and anyone who uses footwraps that are usable by all species as far as I understand

**Comment by tattax** [comment]
(ignore that)

**Comment by Mqiib** [comment]
> > > I mean this doesnt make sense "Most of the foot is not touching the ground" then why does it stop tables and glass? You are opening a can of worms hear, however I will settle if the sound is half as loud as the bare foot one.
> > 
> > 
> > Lizards actually have a different sound than humans when barefoot, being the claws hitting the floor rather than the whole foot slapping down. Footwraps don't totally cover the claws so it'd make sense that they would still be heard while walking. There's no reason for them to be muted at all, and the way it works currently wearing footwraps makes it sound like you're walking around in heavy boots which makes even less sense than what this pr does. As for the glass argument, the soft parts of the foot are still covered while the claws, which are hard and not particularly sensitive, are the only things that would come in contact with the glass.
> 
> What about the other species, its not all about lizards you know?

Footwraps aren't heavy shoes so they shouldn't make heavy boot noises when you're walking around.

**Review by JamieD1 — APPROVED** [review]

File: code/datums/components/footstep.dm [hunk 1]
```diff
@@ -88,17 +88,19 @@
...
+			else
 				if(H.dna.species.special_step_sounds)
 					playsound(T, pick(H.dna.species.special_step_sounds), 50, TRUE)
 				else
 					playsound(T, pick(GLOB.barefootstep[T.barefootstep][1]),
 						GLOB.barefootstep[T.barefootstep][2] * v,
 						TRUE,
 						GLOB.barefootstep[T.barefootstep][3] + e)
+					
+
```
Comment (JamieD1) [inline code/datums/components/footstep.dm]:
```suggestion
```

**Comment by Mqiib** [comment]
image

Shouldn't it be like 
> if((H.shoes && !istype(yadda yadda)) || feetcover) 

since this would be "If shoes and they aren't footwraps OR feet are covered" instead of "If shoes or feet are covered AND shoes are not footwraps" and would avoid possible runtimes when they dont have shoes but have their feet covered?

**Comment by JamieD1** [comment]
> image
> 
> Shouldn't it be like
> 
> > if((H.shoes && !istype(yadda yadda)) || feetcover)
> 
> since this would be "If shoes and they aren't footwraps OR feet are covered" instead of "If shoes or feet are covered AND shoes are not footwraps" and would avoid possible runtimes when they dont have shoes but have their feet covered?

You are correct.

**Review by ToasterBiome — CHANGES_REQUESTED** [review]
@JamieD1 Naughty jamie, now it doesn't build

**Comment by JamieD1** [comment]
> @JamieD1 Naughty jamie, now it doesn't build

Forgot a )

**Review by ToasterBiome — APPROVED** [review]

**Comment by ToasterBiome** [comment]
if this doesn't work it's jamie's fault

**Pull request merged**